pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -484,7 +488,12 @@ def _data_setter(
         -------
         None
         """
+        if not isinstance(X, pd.DataFrame):
+            raise TypeError(
+                "X must be a pandas DataFrame in order to access the columns"
+            )
```

Comment:
I'm viewing this as required in order to access the columns. Changing the type hint makes mypy mad

If there are any other suggestions, let me know

Reply:
I think 99% of the cases people using this package use pandas, so is ok for me.